base/shell/cmd/dir.c

```diff
@@ -1782,6 +1782,10 @@ ResolvePattern(
             pszPatternPart = NULL;
         }
     }
+    else if (_istalpha(pNextDir[0]) && pNextDir[1] == TEXT(':') && pNextDir[2] == TEXT('\0'))
```

Comment:
You should move this as the first case of the if-elseif, before the `if (pszPatternPart == NULL)` test. Otherwise, if you do `dir X:` with `X` a non-existing drive letter, the pszPatternPart will be NULL but then if the `pszPatternPart == NULL` test is first you will have a surprise ^^
Also take the opportunity perhaps to complete the explanatory comment that is above.

Reply:
Please also use `_T()` macro instead of `TEXT()` when you are manipulating characters.

Reply:
OK.